src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectCodeModel.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.LanguageServices.Implementation.CodeModel;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Extensions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS
+{
+    internal sealed class CPSProjectCodeModel : AbstractProjectCodeModel
+    {
+        public CPSProjectCodeModel(CPSProject project, VisualStudioWorkspaceImpl visualStudioWorkspace, IServiceProvider serviceProvider)
+            : base(project, visualStudioWorkspace, serviceProvider)
+        {
+        }
+
+        internal override bool CanCreateFileCodeModelThroughProject(string filePath)
+        {
+            return GetProjectItem(filePath) != null;
```

Comment:
This is pretty expensive for what looks from the outside to be a cheap call. Where is this called?